src/components/ebay-textbox/component.js

```diff
@@ -0,0 +1,70 @@
+import FloatingLabel from "makeup-floating-label";
+
+export default class {
+    onCreate() {
+        this.state = { charCount: 0, value: null };
+    }
+
+    onMount() {
+        this._setupMakeup();
+    }
+
+    onInput(input) {
+        this.state.value = input.value;
+    }
+
+    onUpdate() {
+        this._setupMakeup();
+    }
+
+    focus() {
+        this.getEl("input").focus();
+    }
+
+    _setupMakeup() {
+        // TODO: makeup-floating-label should be updated so that we can remove the event listeners.
+        // It probably makes more sense to just move this functionality into Marko though.
+        if (this.input.floatingLabel) {
+            if (this._floatingLabel) {
+                this._floatingLabel.refresh();
+                this.emit("floating-label-init");
+            } else if (document.readyState === "complete") {
+                if (this.el) {
+                    this._floatingLabel = new FloatingLabel(this.el);
+                    this.emit("floating-label-init");
+                }
+            } else {
+                this.subscribeTo(window).once(
+                    "load",
+                    this._setupMakeup.bind(this)
+                );
+            }
+        }
+    }
+
+    updateCharCount(el) {
+        // debounce
+        clearTimeout(this._charCountTimeout);
+        this._charCountTimeout = setTimeout(() => {
+            this.state.charCount = el.value.length;
+            this.state.value = el.value;
+            this.emit("char-count", {
```

Comment:
I don't like this event name. Maybe `on-char-count-change` or `on-char-change`

Reply:
I second this suggestion. Shouldn't emits of all events be preceded by `on-` at the very least?

Reply:
We already do that in other components. 
<img width="994" alt="image" src="https://github.com/eBay/ebayui-core/assets/1675667/107ef971-5f6d-4dcf-a89e-b3e576fbc696">

Reply:
That component still emits `blur`, `click`, and `focus`. We just change the name in storybook.